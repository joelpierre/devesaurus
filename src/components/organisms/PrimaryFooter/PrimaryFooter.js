import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'gatsby';

import styles from './PrimaryFooter.module.scss';
import Container from '../../core/Container/Container';
import Row from '../../core/Row/Row';
import Flex from '../../core/Flex/Flex';
import TermsMenu from '../../molecules/TermsMenu/TermsMenu';
import Brand from '../../atoms/Brand/Brand';
import SocialMenu from '../../molecules/SocialMenu/SocialMenu';
import Heading from '../../core/Heading/Heading';
import FeaturedWordsList from '../../molecules/FeaturedWordsList/FeaturedWordsList';
import LabelCloud from '../../molecules/LabelCloud/LabelCloud';

const PrimaryFooter = ({ company }) => {
  return (
    <footer
      data-test="component-primary-footer"
      className={styles['primary-footer']}
    >
      <div className={classNames([styles['primary-footer--top']])}>
        <Container>
          <Row>
            <Flex colLg={4} colMd={5}>
              <Heading
                priority={5}
                className={classNames(styles['primary-footer__heading'])}
              >
                About
              </Heading>
              <Heading
                priority={6}
                className={classNames(styles['primary-footer__legal-copy'])}
                innerHTML={false}
              >
                Get real-world definitions to complex dev terms | Supported by{' '}
                <a
                  className={styles['primary-footer__link']}
                  href="http://www.jppdesigns.co.uk"
                >
                  JPPdesigns Web design &amp; Development
                </a>{' '}
                | &copy;{' '}
                {`${company} ${new Date().getFullYear()} | Developed by: Joel Pierre-Powell`}
              </Heading>

              <SocialMenu
                className={classNames('mt-3', styles['primary-footer__social'])}
              />
            </Flex>

            <Flex colLg={3} colMd={6} className="ml-auto">
              <Heading
                priority={5}
                className={classNames(styles['primary-footer__heading'])}
              >
                Featured Words
              </Heading>
              <FeaturedWordsList
                className={classNames(styles['primary-footer__featured-words'])}
              />
            </Flex>

            <Flex colLg={4} colMd={12} className="ml-lg-auto order-lg-1">
              <Heading
                priority={5}
                className={classNames(styles['primary-footer__heading'])}
              >
                Word Tags
              </Heading>
              <div className={styles['primary-footer__tag-cloud-wrapper']}>
                <LabelCloud
                  className={styles['primary-footer__tag-cloud']}
                  taxonomy="word_tag"
                />
              </div>
            </Flex>
          </Row>
        </Container>
      </div>

      <div className={classNames([styles['primary-footer--bottom']])}>
        <Container>
          <Row>
            <Flex
              colLg={5}
              colMd={12}
              className={styles['primary-footer__legal']}
            >
              <Link to="/">
                <Brand
                  left
                  type="text"
                  className={classNames(
                    styles['primary-footer__brand'],
                    'mb-2'
                  )}
                />
              </Link>
            </Flex>

            <Flex colLg={7} colMd={12} className="display-flex">
              <TermsMenu className={styles['primary-footer__terms-menu']} />
            </Flex>
          </Row>
        </Container>
      </div>
    </footer>
  );
};

PrimaryFooter.defaultProps = {
  company: 'Devesaurus',
};

PrimaryFooter.propTypes = {
  company: PropTypes.string,
};

export default PrimaryFooter;
