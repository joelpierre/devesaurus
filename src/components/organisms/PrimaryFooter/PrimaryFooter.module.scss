@import "~sass/core";

.primary-footer {
  $root: &;
  width: 100%;
  background-color: $color-tint-alpha;
  transition: $transition-all-base;
  border-top: .2rem solid $color-tint-gamma;

  &--top {
    padding: 8rem 0 3rem;
    background-color: $color-tint-beta;
  }

  &--bottom {
    padding: 3rem 0;
  }

  &__heading {
    font-weight: bold;
    text-transform: uppercase;
    letter-spacing: 1px;
    display: block;
    width: 100%;
    color: $color-tint-psi;
    position: relative;
    margin-bottom: 3rem !important;

    &:after {
      content: '';
      height: 3px;
      width: 20px;
      background-color: $color-tint-psi;
      position: absolute;
      top: auto;
      bottom: -1rem;
      left: 0;
    }
  }

  &__copy {
    font-size: 1.4rem;
    color: $color-tint-psi;
  }

  &__brand {
    width: 100%;
    max-width: 100%;
    height: 35px;
    padding: 0;
    margin-bottom: 0 !important;

    svg {
      color: $color-tint-omega;
      max-width: 100%;
    }
  }

  &__social {
    margin-bottom: 4rem !important;
  }

  &__featured-words {
    margin-bottom: 4rem !important;
  }

  &__link {
    color: $color-tint-psi;
    font-weight: bold;
  }

  &__legal {
    text-align: center;

    @include tablet-landscape {
      text-align: left;
    }

    &-copy {
      color: $color-tint-psi;
      font-weight: normal;
      font-size: 1.4rem;
      margin-bottom: 2rem;
      line-height: 1.6;

      @include desktop {
        margin-bottom: 0;
      }
    }
  }

  &__terms-menu {
    font-size: 1.4rem;
    justify-content: center;
    margin-left: auto;

    @include tablet-landscape {
      justify-content: flex-start;
    }
    @include desktop {
      justify-content: flex-end;
    }
  }
}
