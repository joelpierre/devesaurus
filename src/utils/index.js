export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
