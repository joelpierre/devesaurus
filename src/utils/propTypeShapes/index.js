import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
