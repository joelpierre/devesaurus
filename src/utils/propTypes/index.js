import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
