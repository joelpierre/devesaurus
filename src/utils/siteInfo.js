import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
