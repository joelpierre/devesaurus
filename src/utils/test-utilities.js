import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
