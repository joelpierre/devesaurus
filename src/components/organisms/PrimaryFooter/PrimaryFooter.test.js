import React from 'react';
import { shallow } from 'enzyme';

import { findByTestAttr } from '../../../utils/test-utilities';

import PrimaryFooter from './PrimaryFooter';

const defaultProps = {};

/**
 * Factory function to create a ShallowWrapper for the PrimaryFooter component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<PrimaryFooter {...setupProps}/>);
};

describe('<PrimaryFooter/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Renders the PrimaryFooter Component without errors', () => {
    const component = findByTestAttr(wrapper, 'VALUE_HERE');
    expect(component.length)
      .toBe(1);
  });


});
