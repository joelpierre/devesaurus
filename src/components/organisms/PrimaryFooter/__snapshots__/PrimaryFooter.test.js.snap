// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryFooter/> should match snapshot 1`] = `
<footer
  className="primary-footer"
  data-test="component-primary-footer"
>
  <div
    className="primary-footer--top"
  >
    <Container
      className={null}
      fluid={false}
    >
      <Row
        className={null}
        column={false}
        reverse={false}
        row={true}
      >
        <Flex
          auto={false}
          colLg={4}
          colMd={5}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <Heading
            className="primary-footer__heading"
            innerHTML={true}
            priority={5}
          >
            About
          </Heading>
          <Heading
            className="primary-footer__legal-copy"
            innerHTML={false}
            priority={6}
          >
            Get real-world definitions to complex dev terms | Supported by
             
            <a
              className="primary-footer__link"
              href="http://www.jppdesigns.co.uk"
            >
              JPPdesigns Web design & Development
            </a>
             
            | Â©
             
            Devesaurus 2019 | Developed by: Joel Pierre-Powell
          </Heading>
          <SocialMenu
            className="mt-3 primary-footer__social"
          />
        </Flex>
        <Flex
          auto={false}
          className="ml-auto"
          colLg={3}
          colMd={6}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <Heading
            className="primary-footer__heading"
            innerHTML={true}
            priority={5}
          >
            Featured Words
          </Heading>
          <FeaturedWordsList
            className="primary-footer__featured-words"
          />
        </Flex>
        <Flex
          auto={false}
          className="ml-lg-auto order-lg-1"
          colLg={4}
          colMd={12}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <Heading
            className="primary-footer__heading"
            innerHTML={true}
            priority={5}
          >
            Word Tags
          </Heading>
          <div
            className="primary-footer__tag-cloud-wrapper"
          >
            <LabelCloud
              className="primary-footer__tag-cloud"
              taxonomy="word_tag"
            />
          </div>
        </Flex>
      </Row>
    </Container>
  </div>
  <div
    className="primary-footer--bottom"
  >
    <Container
      className={null}
      fluid={false}
    >
      <Row
        className={null}
        column={false}
        reverse={false}
        row={true}
      >
        <Flex
          auto={false}
          className="primary-footer__legal"
          colLg={5}
          colMd={12}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <mockConstructor
            to="/"
          >
            <Brand
              center={false}
              className="primary-footer__brand mb-2"
              left={true}
              right={false}
              type="text"
            />
          </mockConstructor>
        </Flex>
        <Flex
          auto={false}
          className="display-flex"
          colLg={7}
          colMd={12}
          lg={false}
          md={false}
          sm={false}
          type="div"
          xl={false}
          xs={false}
        >
          <TermsMenu
            className="primary-footer__terms-menu"
          />
        </Flex>
      </Row>
    </Container>
  </div>
</footer>
`;
